### PR TITLE
Pass dictionary suggestions to the `menu` option function

### PR DESCRIPTION
**Pull request by smithalan92**
At the moment, if you need to use the `menu` function to build a custom menu for whatever reason, you need to duplicate all the work the module does to correctly put spellcheck suggestions into the menu.

This tweak passes the `dictionarySuggestions` Array generated by the module as a 4th param to the `menu` function, so it can be used as is to provide spellcheck in the custom menu template.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -200,7 +200,7 @@ Type: `Function`
...
+The function passed to this option is expected to return [`MenuItem[]`](https://electronjs.org/docs/api/menu-item/). The first argument the function receives is an array of default actions that can be used. These actions are functions that can take an object with a transform property (except for `separator` and `inspect`). The transform function will be passed the content of the action and can modify it if needed. If you use `transform` on `cut`, `copy`, or `paste`, they will convert rich text to plain text. The second argument is [this `params` object](https://electronjs.org/docs/api/web-contents/#event-context-menu). The third argument is the [BrowserWindow](https://electronjs.org/docs/api/browser-window/) the context menu was requested for. The second argument is an Array of pre build menu items for dictionary suggestions. This should be used if you wish to implement spellcheck in your custom menu.
```
Comment (sindresorhus) [inline readme.md:203]: > The second argument is an Array 

I think your meant "fourth" or "last".

Comment (sindresorhus) [reply readme.md:203]: I would also remove `pre build`. It has a typo and is also not needed.

**Comment by sindresorhus** [comment]
You need to update index.d.ts too.

**Comment by sindresorhus** [comment]
I'm also curious why you're using the `menu` option? It should really just be a last option. Maybe there's something that can be done to enable your use-case.

**Comment by smithalan92** [comment]
Hey @sindresorhus , thanks for taking a look. I've pushed those changes as requested.

> I'm also curious why you're using the menu option? It should really just be a last option. Maybe there's something that can be done to enable your use-case.

The reason we need to use the menu option is more to do with how our app is structured, rather than defaults of the module. Basically, Electron loads a web app in a webview, but we load specific version urls to make sure the desktop/web app builds always match. So a desktop version of 1.0.0 would load a url like `/app-version/1.0.0` , however we don't want to expose these version urls to a user, so when links are opened externally, or copied etc.. , we need to rewrite those urls from
`/app-version/1.0.0` to `/app` to keep it nice and simple for an end user, using the `menu` option allows us to do to this.

**Comment by sindresorhus** [comment]
> The reason we need to use the menu option is more to do with how our app is structured, rather than defaults of the module. Basically, Electron loads a web app in a webview, but we load specific version urls to make sure the desktop/web app builds always match. So a desktop version of 1.0.0 would load a url like /app-version/1.0.0 , however we don't want to expose these version urls to a user, so when links are opened externally, or copied etc.. , we need to rewrite those urls from
/app-version/1.0.0 to /app to keep it nice and simple for an end user, using the menu option allows us to do to this.

So you use `menu` because you need the `transform` option for some of the items? Maybe we could add a global transform option or something.

**Pull request merged**